IM_calculation/Advanced_IM/overview_adv_IM_status.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+import argparse
+import glob
+import pandas as pd
+from pathlib import Path
+import sys
+
+ALL_STATUS = [
+    "finished",
+    "not_started",
+    "not_converged",
+    "not_finished",
+    "timed_out",
+    "crashed",
+    "unknown",
+]
+TYPES = ["obs", "sim"]
+
+SIM_DIR_SUFFIX = "Runs/*/*/IM_calc/{}_status.csv"
+OBS_DIR_SUFFIX = "ObservedGroundMotions/IM_calc/*/{}_status.csv"
+
+
+def process(type, model, path, outdir):
+    event_status_files = glob.glob(str(path))
+
+    if len(event_status_files) == 0:
+        print(f"Error:No status files found at {path}")
+        print(f"Check the root_path and model")
+        sys.exit(0)
+
+    event_status_dict = {}
+    for event_status_file in event_status_files:
+        if type == "sim":
+            event_name = Path(event_status_file).parents[2].name
+        else:
+            event_name = Path(event_status_file).parent.name
+        event_status_df = pd.read_csv(event_status_file)
+        event_status_dict[event_name] = {"total": 0}
+        for st in ALL_STATUS:
+            try:
+                count = event_status_df["status"].value_counts()[st]
+            except KeyError:
+                count = 0
```

Comment:
I've done a quick test with value_counts, and it looks like it accepts dict get() notation. So you should be able to do this:`event_status_df["status"].value_counts().get(st, 0)` instead of having the try/except

Reply:
Good idea